packages/account-v2/src/containers/POOForm/FileUploadField.tsx

```diff
@@ -0,0 +1,151 @@
+};
+
+export const FileUploaderField = ({ methodId, paymentMethod, subIndex }: TFileUploadFieldProps) => {
+    const formik = useFormikContext<TProofOfOwnershipFormValue>();
+    const { errors, setFieldError, setFieldValue, values } = formik;
+    const [showBrowseButton, setShowBrowseButton] = useState(
+        !values[paymentMethod]?.[methodId]?.files?.[subIndex]?.name
+    );
+
+    if (!formik) {
+        throw new Error('FileUploaderField must be used within a Formik component');
+    }
+
+    // Create a reference to the hidden file input element
+    const hiddenInputFieldRef = useRef<HTMLInputElement>(null);
+
+    const preventEventBubble = (e: SyntheticEvent) => {
+        e.nativeEvent.preventDefault();
+        e.nativeEvent.stopPropagation();
+        e.nativeEvent.stopImmediatePropagation();
+    };
+
+    const handleChange = async (event: ChangeEvent<HTMLInputElement>) => {
+        preventEventBubble(event);
+        // Check if files exist before proceeding
+        if (!event.target.files || event.target.files.length === 0) {
+            return;
+        }
+        const fileToUpload = await compressImageFiles([event.target.files[0]]);
+        const paymentFileData = [...(values[paymentMethod]?.[methodId]?.files ?? [])];
+        paymentFileData[subIndex] = fileToUpload[0] as TFile;
+        const selectedPaymentMethod = values?.[paymentMethod];
+        if (!selectedPaymentMethod) {
+            return;
+        }
+        selectedPaymentMethod[methodId] = {
+            ...selectedPaymentMethod[methodId],
+            files: paymentFileData ?? [],
+        };
+        await setFieldValue(paymentMethod, { ...selectedPaymentMethod });
+        setShowBrowseButton(!fileToUpload[0]);
+    };
+
+    const handleClick = (event: MouseEvent) => {
+        preventEventBubble(event);
+        hiddenInputFieldRef?.current?.click();
+    };
+
+    const updateError = () => {
+        const paymentMethodError = errors?.[paymentMethod] ?? {};
+        const paymentMethodFileError = (paymentMethodError?.[methodId]?.files as FormikErrors<TFile>[]) ?? {};
+
+        delete paymentMethodFileError?.[subIndex];
+        paymentMethodError[methodId] = {
+            ...(paymentMethodError[methodId] ?? {}),
+            files: paymentMethodFileError,
+        };
+
+        // [TODO] - Need to check the logic for removing the paymentMethodIdentifier
+        if (Object.keys(paymentMethodError[methodId]?.files as object).length === 0) {
+            delete paymentMethodError[methodId]?.paymentMethodIdentifier;
+        }
+        // @ts-expect-error Error is an array
+        setFieldError(paymentMethod, { ...paymentMethodFileError });
+    };
+
+    const handleIconClick = async (e: React.MouseEvent) => {
+        e.nativeEvent.preventDefault();
+        e.nativeEvent.stopPropagation();
+        e.nativeEvent.stopImmediatePropagation();
+
+        if (hiddenInputFieldRef.current && 'value' in hiddenInputFieldRef.current) {
```

Comment:
can we move out this condition outside from if statement ?

Reply:
I didn't understand. Can u give sho, what do u mean?